ksql-engine/src/main/java/io/confluent/ksql/function/UdfLoader.java

```diff
@@ -267,7 +266,15 @@ private void addFunction(final UdfDescription classLevelAnnotation,
           .orElse(param.isNamePresent() ? param.getName() : "");
 
       final String doc = annotation.map(UdfParameter::description).orElse("");
-      return SchemaUtil.getSchemaFromType(type, name, doc);
+      if (annotation.isPresent() && !annotation.get().schema().isEmpty()) {
```

Comment:
What happens if the supplied annotation is incorrect? For example, if someone annotates a `String` field as `BIGINT` for some crazy reason. Is the behavior that the UDF would compile, but you'd only be able to use it by passing in a `BIGINT` field, and then if you did you'd get errors during runtime?

Reply:
As I mentioned in the other PR (#2642) verifying this is non-trivial, so I'm leaving it for future work, but yes at the moment it would fail at runtime. I definitely agree that this is a must-have feature before 5.3 release though.

Reply:
OK...